tests/integrations/object_detection/test_object_detection.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+
+import onnx
+import onnxruntime as ort
+import pytest
+import torch
+
+from tests.integrations.base_tester import (
+    BaseIntegrationManager,
+    BaseIntegrationTester,
+    skip_inactive_stage,
+)
+from tests.integrations.helpers import get_configs_with_cadence
+from tests.integrations.object_detection.object_detection_args import (
+    Yolov5ExportArgs,
+    Yolov5TrainArgs,
+)
+from yolov5.export import load_checkpoint
+from yolov5.val import run as val
+
+
+try:
+    import deepsparse
+except Exception:
+    deepsparse = None
+
+
+METRIC_TO_INDEX = {"map0.5": 2}
+
+
+class ObjectDetectionManager(BaseIntegrationManager):
+
+    command_stubs = {
+        "train": "sparseml.object_detection.train",
+        "export": "sparseml.object_detection.export_onnx",
+    }
+    config_classes = {
+        "train": Yolov5TrainArgs,
+        "export": Yolov5ExportArgs,
+    }
+
+    def capture_pre_run_state(self):
+        super().capture_pre_run_state()
+        if "train" in self.command_types:
+            train_args = self.configs["train"].run_args
+            directory = os.path.dirname(train_args.project)
+            os.makedirs(directory, exist_ok=True)
+            self.save_dir = tempfile.TemporaryDirectory(dir=directory)
+            train_args.project = self.save_dir.name
+
+        if "export" in self.command_types:
+            export_args = self.configs["export"].run_args
+            export_args.weights = export_args.weights or os.path.join(
+                train_args.project, "exp", "weights", "last.pt"
+            )
+        for stage, config in self.configs.items():
+            config.dashed_keywords = True
+
+    def teardown(self):
+        if "train" in self.command_types:
+            self.save_dir.cleanup()
+        # TODO: Add cleanup of export file if export only mode was ran
```

Comment:
is this planned to be implemented before landing? do the other integrations need this?

Reply:
The idea is to have all integration tests to end with 0 net file creation. The cleanup of downloaded Sparsezoo models can be done on the base level. The cleanup of the export file will need to be done on the integrations level. I expected to get both of those done in subsequent PRs (which may land on the same base branch or may be standalone patches, depending on timing).

Reply:
awesome, thanks @KSGulin